### ISSUE TITLE
Mono_Inertial with my own data

### ISSUE THREAD

**Issue by Liu2022** [UZ-SLAMLab/ORB_SLAM3#414]
I want to use my own data to do some experiments. With the help of ROSBAG, I record my camera data and IMU data as "data.bag". The structure of data is almost the same as the structure of Euroc data. But when I run Mono_Inertial and play my "data.bag", the right result cannot be provided. I can only find the picture is playing without creating the map and I think the features of the pictures are even not extracted, which confuses me a lot. Sorry, I am new in the SLAM field and maybe I cannot describe my problem very clearly. I hope someone can help me, thank you very much.

**Comment by tejalbarnwal** [comment]
where do you set time offset? 
is it difference between time stamps of imy data and camera image data?

**Comment by geturin** [comment]
yes, it is difference between time stamps of imu data and camera image data
ORB_SLAM can't set it, so you should change the camera's time stamp

**Comment by tejalbarnwal** [comment]
How can I manupilate the timestamps in a previously recorded bag file?
I apologize for asking such nood questions.
Also, any source from where I can learn about these stuffs about ORB-SLAM 3? these crucial stuffs like time stamp change and tuning params like lapping area, camera.bf and th.depth

**Comment by geturin** [comment]
https://github.com/UZ-SLAMLab/ORB_SLAM3/issues/78
you can looking for other similar questions in issues

about timestamp,
I subsucriber raw image data,  new time stamp = raw time stamp + rospy.Duration("time offset")
http://wiki.ros.org/rospy/Overview/Time

not very good,maybe have the another better way
